src/backend/storage/buffer/localbuf.c

```diff
@@ -279,35 +292,45 @@ LocalBufferAlloc(SMgrRelation smgr, ForkNumber forkNum, BlockNumber blockNum,
 					 errmsg("no empty local buffer available")));
 	}
 
+	/* Remotexact - bufHdr is changed after the above loop so we need to update remote_bufHdr */
+	remote_bufHdr = LocalBufHdrGetRemoteDesc(bufHdr);
+
 	/*
 	 * this buffer is not referenced but it might still be dirty. if that's
 	 * the case, write it out before reusing it!
 	 */
 	if (buf_state & BM_DIRTY)
 	{
-		SMgrRelation oreln;
-		Page		localpage = (char *) LocalBufHdrGetBlock(bufHdr);
-
-		/* Find smgr relation for buffer */
-		if (am_wal_redo_postgres && MyBackendId == InvalidBackendId)
-			oreln = smgropen(bufHdr->tag.rnode, MyBackendId, RELPERSISTENCE_PERMANENT, UNKNOWN_REGION);
-		else
-			oreln = smgropen(bufHdr->tag.rnode, MyBackendId, RELPERSISTENCE_TEMP, UNKNOWN_REGION);
-
-		PageSetChecksumInplace(localpage, bufHdr->tag.blockNum);
+		/*
+		 * Remotexact
+		 * Just throw away the page if it comes from a remote relation
```

Comment:
I would reword it as write localbuf to disk only if it does not belong to a remote relation.

Reply:
Done

Reply:
Actually, I can just skip the remote pages in the clock sweep run, and the `no empty local buffer available` error will be triggered when we don't have enough buffer